shared-bindings/bleio/Scanner.c

```diff
@@ -29,140 +30,102 @@
+//|    entries = scanner.scan(2.5)  # Scan for 2.5 seconds
 //|
 
 //| .. class:: Scanner()
 //|
 //|   Create a new Scanner object.
 //|
-
-//|   .. attribute:: interval
-//|
-//|     The interval (in ms) between the start of two consecutive scan windows.
-//|     Allowed values are between 10ms and 10.24 sec.
-//|
-
-//|   .. attribute:: window
-//|
-//|     The duration (in ms) in which a single BLE channel is scanned.
-//|     Allowed values are between 10ms and 10.24 sec.
-//|
-
-//|   .. method:: scan(timeout)
-//|
-//|     Performs a BLE scan.
-//|
-//|     :param int timeout: the scan timeout in ms
-//|     :returns: advertising packets found
-//|     :rtype: list of :py:class:`bleio.ScanEntry`
-//|
-STATIC void bleio_scanner_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
-    bleio_scanner_obj_t *self = MP_OBJ_TO_PTR(self_in);
-
-    mp_printf(print, "Scanner(interval: %d window: %d)", self->interval, self->window);
-}
-
 STATIC mp_obj_t bleio_scanner_make_new(const mp_obj_type_t *type, size_t n_args, const mp_obj_t *all_args, mp_map_t *kw_args) {
     mp_arg_check_num(n_args, kw_args, 0, 0, false);
 
     bleio_scanner_obj_t *self = m_new_obj(bleio_scanner_obj_t);
     self->base.type = type;
 
-    self->interval = DEFAULT_INTERVAL;
-    self->window = DEFAULT_WINDOW;
+    common_hal_bleio_scanner_construct(self);
 
     return MP_OBJ_FROM_PTR(self);
 }
 
-STATIC mp_obj_t bleio_scanner_get_interval(mp_obj_t self_in) {
-    bleio_scanner_obj_t *self = MP_OBJ_TO_PTR(self_in);
-
-    return mp_obj_new_int(self->interval);
-}
-MP_DEFINE_CONST_FUN_OBJ_1(bleio_scanner_get_interval_obj, bleio_scanner_get_interval);
-
-static mp_obj_t bleio_scanner_set_interval(mp_obj_t self_in, mp_obj_t value) {
-    bleio_scanner_obj_t *self = MP_OBJ_TO_PTR(self_in);
-
-    self->interval = mp_obj_get_int(value);
-
-    return mp_const_none;
-}
-STATIC MP_DEFINE_CONST_FUN_OBJ_2(bleio_scanner_set_interval_obj, bleio_scanner_set_interval);
-
-const mp_obj_property_t bleio_scanner_interval_obj = {
-    .base.type = &mp_type_property,
-    .proxy = { (mp_obj_t)&bleio_scanner_get_interval_obj,
-               (mp_obj_t)&bleio_scanner_set_interval_obj,
-               (mp_obj_t)&mp_const_none_obj },
-};
-
-STATIC mp_obj_t scanner_scan(mp_obj_t self_in, mp_obj_t timeout_in) {
-    bleio_scanner_obj_t *self = MP_OBJ_TO_PTR(self_in);
-    const mp_int_t timeout = mp_obj_get_int(timeout_in);
+//|   .. method:: scan(timeout, \*, interval=0.1, window=0.1)
+//|
+//|     Performs a BLE scan.
+//|
+//|     :param float timeout: the scan timeout in seconds
+//|     :param float interval: the interval (in seconds) between the start of two consecutive scan windows
+//|        Must be in the range 0.0025 - 40.959375 seconds.
+//|     :param float window: the duration (in seconds) to scan a single BLE channel.
+//|        window must be <= interval.
+//|     :returns: advertising packets found
+//|     :rtype: list of :py:class:`bleio.ScanEntry`
```

Comment:
I removed the attribute that returns the saved list, so now the caller of `scan()` has exclusive ownership of the list. So it doesn't need to be immutable: the caller is free to change it. But I'll change the doc to say it just returns an iterable, since it's better to be general: other implementations might not return a list.